cmd/podman/rmi.go

```diff
@@ -121,22 +122,21 @@ func rmiCmd(c *cliconfig.RmiValues) error {
 		for _, i := range images {
 			newImage, err := runtime.NewImageFromLocal(i)
 			if err != nil {
-				fmt.Fprintln(os.Stderr, err)
+				if lastError != nil {
+					if errors.Cause(lastError) != image.ErrNoSuchImage {
+						failureCnt++
+					}
+					fmt.Fprintln(os.Stderr, lastError)
+				}
+				lastError = err
 				continue
 			}
 			removeImage(newImage)
 		}
 	}
 
-	// If the user calls remove all and there are none, it should not be a
-	// non-zero exit
-	if !deleted && removeAll {
```

Comment:
Did we lose this logic? Or are we intentionally changing --all to return 1 if there are no images?

I'm still of the opinion that `rmi --all` on 0 images is a 0 return

Reply:
Seems to still work.

```
./bin/podman rmi --all 
[dwalsh@localhost libpod]$ echo $?
0
```

Reply:
Huh. Alright, fair enough.